app/src/main/java/com/asfoundation/wallet/topup/address/BillingAddressTopUpPresenter.kt

```diff
@@ -1,23 +1,34 @@
 package com.asfoundation.wallet.topup.address
 
-import com.asfoundation.wallet.ui.iab.Navigator
+import com.asfoundation.wallet.billing.analytics.BillingAnalytics
+import com.asfoundation.wallet.topup.TopUpAnalytics
 import io.reactivex.Scheduler
 import io.reactivex.disposables.CompositeDisposable
 
 class BillingAddressTopUpPresenter(private val view: BillingAddressTopUpView,
+                                   private val data: BillingAddressTopUpData,
                                    private val disposables: CompositeDisposable,
                                    private val viewScheduler: Scheduler,
-                                   private val navigator: Navigator) {
+                                   private val navigator: BillingAddressTopUpNavigator,
+                                   private val topUpAnalytics: TopUpAnalytics) {
 
   fun present() {
+    initializeView()
     handleSubmitClicks()
   }
 
+  private fun initializeView() {
+    view.initializeView(data.data, data.fiatAmount, data.fiatCurrency, data.shouldStoreCard,
+        data.isStored)
```

Comment:
The main reason is because the view should only have what is needed to render the view and it should be descriptive (and it should be fairly "dumb")... In this case I don't think it matters much eitherway, but there's cases where the 'data' object has a lot more info than what is needed for the view (e.g. see CarrierVerifyPresenter, initializeView, l43 and CarrierVerifyData). The only argument I have in this case is for the sake of consistency.